src/components/editor/EditorAreaMenu.tsx

```diff
@@ -71,6 +71,27 @@ export default function Banner(props: { [x: string]: any }) {
 		}
 	};
 
+	const handleHTMLSave = (htmlToSave: string | undefined) => {
+		if (htmlToSave) {
+			const blob = new Blob([htmlToSave], { type: "text/html" });
+	
+			const fileName = "ed-eater.html";
+	
+			const url = URL.createObjectURL(blob);
+
+			const a = document.createElement("a");
+			a.href = url;
+			a.download = fileName;
+			document.body.appendChild(a);
+			a.click();
+
+			URL.revokeObjectURL(url);
+			document.body.removeChild(a);
```

Comment:
This downloading logic is nontrivial and has been used once before somewhere iirc. Would it be better to export the downloading logic to some function? Although it is just repeated twice but adds a layer of complexity IMO.

Reply:
You're right. I'll abstract it to a 'download' function.